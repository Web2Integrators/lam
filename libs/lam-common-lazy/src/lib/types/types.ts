/**
 * The desktop application's possible UI modes.
 */
export enum UiMode { Normal, Embedded }

/**
 * The environment object for the desktop application.
 * It is created from the CLI arguments and version.ts.
 */
export interface Environment {
  startX?: number;
  startY?: number;
  width?: number;
  height?: number;
  uiMode: UiMode;
  startHidden: boolean;
  openDevTools: boolean;
  backendLogging: boolean;
  pdeUrl: string;
  defaultToolAddress: string;
  backendUrl?: string;
  ctuUrl?: string;
  banner?: string;
  versionInfo: {
    dirty?: boolean | null;
    raw?: string | null;
    hash?: string | null;
    distance?: any;
    tag?: any;
    semver?: any;
    suffix?: string | null;
    semverString?: string | null;
    version?: string | null;
  };
}
