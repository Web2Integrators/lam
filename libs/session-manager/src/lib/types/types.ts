export interface LoginCredentials {
  username: string;
  password: string;
}

export interface StoredRoute {
  url?: string;
  params: {
    toolId?: string;
    resourceName?: string;
  };
}

export interface MachineResource {
  displayName: string;
  machineResourceName: string;
  locked: boolean;
  lockInformation: string;
  offline: boolean;
  pmImage: string;
  modelName?: string;
}

export interface HeartbeatResponse {
  remainingInactivityTime: number; // the time in s until the server will time out
  arbitration: Resource[]; // the list of available resources
}

export interface Resource {
  displayName: string;
  lockStatus: string;
  lockType: string;
  lockInformation: string;
  overrideOption: string;
  machineResourceName: string;
  state: string;
}

export interface BackendConfigOptions {
  arbitrationEnabled: boolean;
  endpointEditorEnabled: boolean;
  hydraEditorEnabled: boolean;
}

export interface Session {
  sessionID?: string;
  toolID?: string;
  hostname?: string;
  address?: string;
  sessionName?: string;
  workerPort?: number;
  sessionInactivityTime?: number;
  sessionHeartbeatInterval?: number;
}

export interface PdeConfig {
  defaultToolAddress: string;
  devBackendUrl?: string;
  devCtuUrl?: string;
  banner?: string;
}

export interface HostImage {
  RecipeEditorArbitration: string;
}

export interface PMImage {
  // These commented props are not currently used but do involve existing things in PDE (e.g. MMP):
  // MMPControl: boolean;
  // ModuleType: string;
  // ModuleLabel: string;
  HydraControllerInstalled?: boolean;
  HydraControllerMode?: string;
  OESType?: string;
}

export interface ImageOptions {
  [imageName: string]: HostImage | PMImage; // Keys are "<name>Image", e.g. "HostImage", "PM1Image".
}

export interface MachineConfiguration {
  // MachineConfiguration is a large object. These are only the properties we need.
  imageOptions: ImageOptions;
}

export interface ConfigurationResponse {
  machineConfiguration: MachineConfiguration;
}

/** Message sent from 2300 in a Server Sent Events data field. */
export interface FocusChangedMessage {
  resourceName: string; // e.g. "PM1"
}

/** Possible types sent from 2300 in a Server Sent Events type field. */
export enum SseType {
  FocusChanged = 'FocusChanged',
  HydraTemplateDataChanged = 'HydraTemplateDataChanged',
}

export enum WizardStep {
  Session,
  Login,
  Arbitration,
  Complete,
}

export enum ResetConnectionMode {
  Normal,
  FatalError,
}



export type InvalidState = 'PENDING' | undefined;
