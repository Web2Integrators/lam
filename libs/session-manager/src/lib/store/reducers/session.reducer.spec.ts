import { reducer } from './session.reducer';
import * as fromSession from './session.reducer';
import { Session } from '../../types/types';


describe('SessionsReducer', () => {
  const session: Session = {
    sessionID :'2'
  };

  const initialState: fromSession.State = {
    session: session
  };

  describe('undefined action', () => {
    it('should return the default state', () => {
      const result = reducer(undefined, {} as any);
      expect(result).toMatchSnapshot();
    });
  });

  describe('Selectors', () => {
    describe('selectId', () => {
      it('should return session', () => {
        const result = fromSession.getSession({...initialState});
        expect(result).toMatchSnapshot();
      });
    });
  });

});
