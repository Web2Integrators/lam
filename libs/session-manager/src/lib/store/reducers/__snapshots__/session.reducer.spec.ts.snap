// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionsReducer Selectors selectId should return session 1`] = `
Object {
  "sessionID": "1",
}
`;

exports[`SessionsReducer undefined action should return the default state 1`] = `
Object {
  "session": Object {
    "sessionID": "1",
  },
}
`;
